packages/lit-dev-content/site/articles/article/lit-cheat-sheet.md

```diff
@@ -0,0 +1,1118 @@
+  - `element.appendChild(element)`
+  - `element.innerHTML = '<my-element></my-element>'`
+  - etc.
+- Can run multiple times, but a good place to set up event listeners
+
+#### `willUpdate`
+
+- Runs on the server – do **NOT** access the DOM or browser APIs
+- A good place to set up properties that depend on other properties
+
+#### `update`
+
+- A good place to update properties that depend on other properties that depend on DOM
+
+#### `render`
+
+- Runs on the server - do NOT access the DOM or browser APIs
+
+#### `firstUpdated`
+
+- Runs after the first render
+- A good place to perform initializations that require access to the component's
+  rendered DOM
+
+#### `updated`
+
+- A good place to do some updates that require the component's rendered DOM or
+  to update properties that depend on the rendered DOM
+- Avoid setting reactive properties in this lifecycle as doing so may trigger
+  unnecessary re-renders. Try to do them in `willUpdate` or `update` if
+  possible.
+
+#### `disconnectedCallback`
+
+- Is called *AFTER* the element is removed from the DOM
+- A good place to clean up event listeners
+
+### Waiting for an update
+
+All Lit elements have asynchronous lifecycles. You need to wait for the `updateComplete` promise to resolve before you can be sure that the element has finished updating its DOM.
+
+{% playground-ide "articles/lit-cheat-sheet/update-complete", true %}
+
+**Related Documentation & Topics:**
+
+- [Lifecycle](/docs/components/lifecycle/)
+- [updateComplete()](/docs/components/lifecycle/#updatecomplete)
+- [requestUpdate()](/docs/components/lifecycle/#requestUpdate)
+- [Getting an Element Reference](#getting-an-element-reference)
+- [Getting an Element Reference After Update](#getting-an-element-reference-after-update)
+
+### Asynchronous Tasks
+
+If you need to perform an asynchronous task in a Lit Element. You may want to
+use the `@lit/task` package. It handles marrying basics of managing asynchronous
+and the Lit lifecycle.
+
+The following example fetches a Pokemon by ID from the
+[PokeAPI](https://pokeapi.co/) based on pokemon name. To do so you:
+
+1. Initialize the task with `new Task(...)`
+2. Task is a Reactive Controller, so you pass it a reference to the Reactive
+   Element (`this`)
+3. Write an async function that fetches and returns the data
+4. Give Task a function that will return the reactive properties that Task
+   relies on
+5. Render all the states of the Task in the `render()` method with
+   `Task.prototype.render()`
+
+{% playground-ide "articles/lit-cheat-sheet/task", true %}
+
+**Related Documentation & Topics:**
+
+- [Asynchronous Tasks](/docs/data/task/)
+- [Lit Labs – Task](https://www.youtube.com/watch?v=niWKuGhyE0M) (video)
+  - NOTE: This video was published before Task had graduated from Labs.
+- [Reactive Controllers – Asynchronous Tasks](/docs/composition/controllers/#asynchronous-tasks)
+
+## Events
+
+### Adding listeners to host element or globally
```

Comment:
I don't think we should encourage setting up event listeners on host in `connectedCallback`. This should generally be done in `constructor` and doesn't need cleanup. Promoting this potentially feeds FUD about memory leaks. Recommending changing the text here and the associated example to focus on listeners to external elements.